File: android/GeoQuiz/app/src/main/java/com/cooldude/android/geoquiz/CheatActivity.java
package com.cooldude.android.geoquiz;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class CheatActivity extends AppCompatActivity {

    private static final String EXTRA_ANSWER_IS_TRUE = "com.cooldude.android.geoquiz.answer_is_true";
    private static final String EXTRA_ANSWER_SHOWN = "com.cooldude.android.geoquiz.answer_shown";
    private static final String KEY_INDEX = "index";

    private boolean mAnswerIsTrue;
    private TextView mAnswerTextView;
    private TextView mVersionTextView;
    private Button mShowAnswerButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_cheat);

        mAnswerTextView = (TextView)findViewById(R.id.answer_text_view);

        if(savedInstanceState != null){
            mAnswerIsTrue = savedInstanceState.getBoolean(KEY_INDEX);
            if(mAnswerIsTrue){
                mAnswerTextView.setText(R.string.true_button);
            }
            else{
                mAnswerTextView.setText(R.string.false_button);
            }
            setAnswerShownResult(mAnswerIsTrue);
        }
        else {
            mAnswerIsTrue = getIntent().getBooleanExtra(EXTRA_ANSWER_IS_TRUE, false);
        }

        mVersionTextView = (TextView)findViewById(R.id.build_version);
        mVersionTextView.setText(getResources().getString(R.string.build_version, android.os.Build.VERSION.SDK_INT));

        mShowAnswerButton = (Button)findViewById(R.id.show_answer_button);
        mShowAnswerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(mAnswerIsTrue){
                    mAnswerTextView.setText(R.string.true_button);
                }
                else{
                    mAnswerTextView.setText(R.string.false_button);
                }

                setAnswerShownResult(true);
            }
        });
    }

    private void setAnswerShownResult(boolean isAnswerShown){
        Intent data = new Intent();
        data.putExtra(EXTRA_ANSWER_SHOWN, isAnswerShown);
        setResult(RESULT_OK, data);
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        overridePendingTransition(R.anim.slide_in_down, R.anim.slide_out_down);
    }

    public static boolean wasAnswerShown(Intent result){
        return result.getBooleanExtra(EXTRA_ANSWER_SHOWN, false);
    }

    public static Intent newIntent(Context packageContext, boolean answerIsTrue){
        Intent intent = new Intent(packageContext, CheatActivity.class);
        intent.putExtra(EXTRA_ANSWER_IS_TRUE, answerIsTrue);
        return intent;
    }


    @Override
    protected void onSaveInstanceState(Bundle savedInstanceState){
        super.onSaveInstanceState(savedInstanceState);
        savedInstanceState.putBoolean(KEY_INDEX, mAnswerIsTrue);
    }
}

File: android/Geoquiz-Kotlin/app/src/main/java/com/me/geoquiz/CheatActivity.kt
package com.me.geoquiz

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.ViewAnimationUtils
import android.widget.Button
import android.widget.TextView

class CheatActivity : AppCompatActivity() {
    private val KEY_NUM_CHEATED = 0
    private var answerIsTrue: Boolean? = null
    private var answerView: TextView? = null
    private var answerButton: Button? = null


    companion object {
        private const val EXTRA_ANSWER_TRUE = "com.me.geoquiz.answer_is_true"
        private const val EXTRA_ANSWER_SHOWN = "com.me.geoquiz.answer_is_shown"

        fun newIntent(context: Context, answerIsTrue: Boolean): Intent {
            val intent = Intent(context, CheatActivity::class.java)
            intent.putExtra(EXTRA_ANSWER_TRUE, answerIsTrue)
            return intent
        }

        fun wasAnswerShown(result: Intent): Boolean {
            return result.getBooleanExtra(EXTRA_ANSWER_SHOWN, false)
        }
    }

    private fun setAnswerShownResult(isAnswerShown: Boolean) {
        val data = Intent()
        data.putExtra(EXTRA_ANSWER_SHOWN, isAnswerShown)
        setResult(RESULT_OK, data)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_cheat)

        answerIsTrue = intent.getBooleanExtra(EXTRA_ANSWER_TRUE, false)
        answerView = findViewById(R.id.answerText)
        answerButton = findViewById(R.id.answerButton)
        answerButton?.setOnClickListener {
            if (answerIsTrue == true) {
                answerView?.setText(R.string.true_button)
            } else {
                answerView?.setText(R.string.false_button)
            }

            setAnswerShownResult(true)

            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                val cx = it.width / 2
                val cy = it.height / 2
                val radius: Float = it.width.toFloat()
                val anim: Animator = ViewAnimationUtils.createCircularReveal(it, cx, cy, radius, 0.toFloat())
                anim.addListener(object : AnimatorListenerAdapter() {
                    override fun onAnimationEnd(animation: Animator?) {
                        it.visibility = View.INVISIBLE
                    }
                })
                anim.start()
            } else {
                it.visibility = View.INVISIBLE
            }
        }

    }
}
